src/boot/efi/boot.c

```diff
@@ -897,6 +898,34 @@ static VOID config_add_entry(Config *config, ConfigEntry *entry) {
         config->entries[config->entry_count++] = entry;
 }
 
+// TODO: very unsure of this function, looks dangerous.
+static VOID config_remove_by_id_startswith(Config *config, CHAR16 *id_startswith) {
```

Comment:
This whole function can be simplified with some pointer arithmetic:

```
removed = 0;
for (i = j = config->entries; i < config->entries + config->entries_count; i++) {
    if (startswith(entry->id, id_startswith)) {
        FreePool(i);
         removed++;
   } else {
         if (i != j)
               *j = *i;
          j++;
          continue;
    }
}
config->entries_count -= removed;
```

Reply:
I sometimes feel like lots of things could be simplified with pointer arithmetic but I'd like to avoid it has much as possible out of performance critical code.

If this is an important issue I am however prepared to change it. As you wish.

Reply:
Yes please, I don't like looking at that function when it can be made so much easier. Actually, it doesn't have to be pointer arithmetic, it works with indices too. Pointer arithmetic here would just make it much more pleasant to look at.